lib/experimental/interactivity-api/class-wp-interactivity-initial-state.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * WP_Interactivity_Initial_State class
+ *
+ * @package Gutenberg
+ * @subpackage Interactivity API
+ */
+
+if ( class_exists( 'WP_Interactivity_Initial_State' ) ) {
+	return;
+}
+
+/**
+ * Manages the initial state of the Interactivity API store in the server and
+ * its serialization so it can be restored in the browser upon hydration.
+ *
+ * @package Gutenberg
+ * @subpackage Interactivity API
+ */
+class WP_Interactivity_Initial_State {
+	/**
+	 * Map of initial state by namespace.
+	 *
+	 * @var array
+	 */
+	private static $initial_state = array();
+
+	/**
+	 * Get state from a given namespace.
+	 *
+	 * @param string $store_ns Namespace.
+	 *
+	 * @return array The requested state.
+	 */
+	public static function get_state( $store_ns ) {
+		return self::$initial_state[ $store_ns ] ?? array();
+	}
+
+	/**
+	 * Merge data into the state with the given namespace.
+	 *
+	 * @param string $store_ns Namespace.
+	 * @param array  $data      State to merge.
+	 *
+	 * @return void
+	 */
+	public static function merge_state( $store_ns, $data ) {
+		self::$initial_state[ $store_ns ] = array_replace_recursive(
+			self::get_state( $store_ns ),
+			$data
+		);
+	}
+
+	/**
+	 * Get store data.
+	 *
+	 * @return array
+	 */
+	public static function get_data() {
+		return self::$initial_state;
+	}
+
+	/**
+	 * Reset the initial state.
+	 */
+	public static function reset() {
+		self::$initial_state = array();
```

Comment:
Should there be a warning like a `_doing_it_wrong` to prevent external plugins to reset the initial state?

Do we want this function to be available to developers?

Reply:
We'll remove that method once we refactor it to get rid of the statics, so don't worry about it 🙂

Reply:
To be honest, the only thing we want developers to use is `wp_initial_state()`. 😅 Other stuff is meant to be used internally.

Is `_doing_it_wrong()` the way to prevent developers from calling those functions? 👀

Reply:
> Is _doing_it_wrong() the way to prevent developers from calling those functions? 👀

That won't prevent it, I think it just informs. I guess not using statics is the best way to do it.